frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -573,9 +587,28 @@ function ReaderHighlight:_resetHoldTimer(clear)
     end
 end
 
-function ReaderHighlight:onHold(arg, ges)
+function ReaderHighlight:getRegionalZoomCenter(pos)
+    if not self.dimen then
+        self.dimen = self.ui.dimen
+    end
+    local page_size = self.ui.document:getNativePageDimensions(pos.page)
+    local margin = self.ui.document.configurable.page_margin * Screen:getDPI()
+    local zoom = 2*self.dimen.w / page_size.w
+    zoom = zoom/(1 + 3*margin/zoom/page_size.w)
+    local x_ratio = (pos.x/ page_size.w) -- x center ratio
+    local y_ratio = (pos.y / page_size.h) -- y center ratio
+    return zoom, x_ratio, y_ratio
+end
+
+
+function ReaderHighlight:onHold(_, ges)
     -- disable hold gesture if highlighting is disabled
-    if self.view.highlight.disabled then return false end
+    if self.view.highlight.disabled then
+        if self.view.highlight.magnifier then
+           self:onTouchZoom(nil, ges) 
```

Comment:
This could probably be rewritten as 
```lua
UIManager:broadcastEvent(Event:new("TouchZoom", ges))
```

Reply:
No need to, when the single and only onTouchZoom handler is in here.

Reply:
I was thinking that @yparitcher's dispatcher can use this as an event, either is fine imo.

Reply:
Yes, he can - and should be able to just by having it named here `onTouchZoom` - use that via broadcastEvent elsewhere - but here, we should call it directly `self:onTouchZoom()` for clarity.
(Although given that this functions needs a gesture with x/y, it's probably not a candidate for profiles or multiswipe gestures.)

Reply:
Multiswipes work with coordinates. No problem there.